.github/workflows/pre-commit.yml

```diff
@@ -11,19 +11,10 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
-      - name: Set up Python
-        uses: actions/setup-python@v5
-        with:
-          python-version: "3.10"
-
-      - name: Install dependencies
-        run: |
-          python -m pip install -U pip
-          pip install -U pre-commit
-          pre-commit install --install-hooks
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
 
       - name: pre-commit
-        run: |
-          pre-commit run --all-files
+        run: uvx --with pre-commit-uv --python 3.10 -- pre-commit run --all-files
```

Comment:
```suggestion
        run: |
        uvx --with pre-commit-uv -- pre-commit run --all-files
```

Reply:
is this the best syntax for uvx? --with comes first and we need -- before pre-commit?

Reply:
It doesn't matter when you pass `--with X`. I just find the command slightly more readable because it tells you where the executable will be coming from. On the other hand, I like your suggestion to replace "uv tool install --with [package] [tool]" with "uv tool install [tool] --with [package]".

The `--` is not needed, but I like it because it's a nice visual separation between the args for uv and the ones for whatever you're running.

Reply:
Also, I think sometimes `--with` and `--from` might be interchangeable, but I might be wrong.